module/zcommon/zfs_prop.c

```diff
@@ -51,7 +51,11 @@ const char *zfs_userquota_prop_prefixes[] = {
 	"userused@",
 	"userquota@",
 	"groupused@",
-	"groupquota@"
+	"groupquota@",
+	"userdnused@",
```

Comment:
I think that "dn" or "dnode" shouldn't be exposed in the user interface.  Instead, we should use user-visible concepts like files or directories.  If we are to introduce a new concept to users/sysadmins, I think it should be "objects" rather than "dnodes".  An object being the conceptual entity, which has some on-disk metadata including the dnode.